src/main/java/com/revature/controllers/TripController.java

```diff
@@ -0,0 +1,140 @@
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/trips")
+@CrossOrigin
+@Api(tags = {"Trips"})
+public class TripController {
+    @Autowired
+    private TripService tripService;
+
+    /**
+     * HTTP GET method (/trips)
+     *
+     * @return A list of all the trips.
+     */
+    @ApiOperation(value = "Return all trips", tags = {"Trip"})
+    @GetMapping
+    public ResponseEntity<List<Trip>> getTrips() {
+        List<Trip> trips = tripService.getTrips();
+
+        if (trips.size() != 0) return ResponseEntity.ok(trips);
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP GET method (/trips/{number})
+     *
+     * @param id represent the trip's ID.
+     * @return A trip that matches the ID.
+     */
+    @ApiOperation(value = "Return trip by ID", tags = {"Trip"})
+    @GetMapping("/{id}")
+    public ResponseEntity<Trip> getTripById(@PathVariable("id") int id) {
+        Optional<Trip> trip = Optional.ofNullable(tripService.getTripById(id));
+
+        if (trip.isPresent()) return ResponseEntity.ok(trip.get());
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP GET method (/trips/driver/{driveId})
+     *
+     * @param driverId represents the driver's ID.
+     * @return A list of trips matching the driver's ID.
+     */
+    @ApiOperation(value = "Returns trips by driver ID", tags = {"Driver", "Trip"})
+    @GetMapping("/driver/{driverId}")
+    public ResponseEntity<List<Trip>> getTripsByDriverId(@PathVariable("driverId") int driverId) {
+        List<Trip> trips = tripService.getTripsByDriverId(driverId);
+
+        if (trips.size() != 0) return ResponseEntity.ok(trips);
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP GET method (/trips/rider/{riderId})
+     *
+     * @param riderId represents the rider's ID.
+     * @return A list of trips matching the rider's ID.
+     */
+    @ApiOperation(value = "Returns trips by rider ID", tags = {"Rider", "Trip"})
+    @GetMapping("/rider/{riderId}")
+    public ResponseEntity<List<Trip>> getTripsByRiderId(@PathVariable("riderId") int riderId) {
+        List<Trip> trips = tripService.getTripsByRiderId(riderId);
+
+        if (trips.size() != 0) return ResponseEntity.ok(trips);
+
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * HTTP POST method (/trips)
+     * @param trip represents the new Trip object being sent.
+     * @return The newly created object along with a 201 code.
+     */
+    @ApiOperation(value = "Adds a new trip")
+    @PostMapping
+    public ResponseEntity<Trip> addTrip(@Valid @RequestBody Trip trip) {
```

Comment:
This has the same flaw: You create entities when it is a bad request.

Reply:
Resolved.